File: app/src/main/java/com/technicaltest/technicaltest/app/services/retrofit/RetrofitClient.java
package com.technicaltest.technicaltest.app.services.retrofit;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitClient {

    public DSPIdentityApi provideRetrofitClientDspIdentity() {
        OkHttpClient.Builder httpClientBuilder = new OkHttpClient.Builder().cache(null);
        HttpLoggingInterceptor httpLoggingInterceptor = new HttpLoggingInterceptor();
        httpLoggingInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
        httpClientBuilder.addInterceptor(httpLoggingInterceptor);

        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl(BuildConfig.DSP_IDENTITY_BASE_URL)
                .client(httpClientBuilder.build())
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .build();

        return retrofit.create(DSPIdentityApi.class);
    }

    public DSPOauthApi provideRetrofitClientDspOauth() {
        OkHttpClient.Builder httpClientBuilder = new OkHttpClient.Builder().cache(null);
        HttpLoggingInterceptor httpLoggingInterceptor = new HttpLoggingInterceptor();
        httpLoggingInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
        httpClientBuilder.addInterceptor(httpLoggingInterceptor);

        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl(BuildConfig.DSP_OAUTH_BASE_URL)
                .client(httpClientBuilder.build())
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .build();

        return retrofit.create(DSPOauthApi.class);
    }

    public DSPIdpApi provideRetrofitClientIdp() {
        OkHttpClient.Builder httpClientBuilder = new OkHttpClient.Builder().cache(null);
        HttpLoggingInterceptor httpLoggingInterceptor = new HttpLoggingInterceptor();
        httpLoggingInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
        httpClientBuilder.addInterceptor(httpLoggingInterceptor);

        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl(BuildConfig.DSP_IDP_BASE_URL)
                .client(httpClientBuilder.build())
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .build();

        return retrofit.create(DSPIdpApi.class);
    }

    public DSPApi provideRetrofitClientDspApi() {
        OkHttpClient.Builder httpClientBuilder = new OkHttpClient.Builder().cache(null);
        HttpLoggingInterceptor httpLoggingInterceptor = new HttpLoggingInterceptor();
        httpLoggingInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
        httpClientBuilder.addInterceptor(httpLoggingInterceptor);
        httpClientBuilder.addInterceptor(new RetrofitInterceptor());

        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl(BuildConfig.DSP_API_BASE_URL)
                .client(httpClientBuilder.build())
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .build();

        return retrofit.create(DSPApi.class);
    }
}

File: app/src/main/java/com/technicaltest/technicaltest/app/services/retrofit/RetrofitClient.kt
package com.technicaltest.technicaltest.app.services.retrofit

import com.technicaltest.technicaltest.BuildConfig
import com.technicaltest.technicaltest.dataSources.meep.Interfaces.MeetApi
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitClient {

    fun provideRetrofitClientMeetApi(): MeetApi {
        val httpClientBuilder = OkHttpClient.Builder().cache(null)
        val httpLoggingInterceptor = HttpLoggingInterceptor()
        httpLoggingInterceptor.level = HttpLoggingInterceptor.Level.HEADERS
        httpClientBuilder.addInterceptor(httpLoggingInterceptor)
        val retrofit: Retrofit = Retrofit.Builder()
            .baseUrl(BuildConfig.MEET_API_BASE_URL)
            .client(httpClientBuilder.build())
            .addConverterFactory(GsonConverterFactory.create())
            .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
            .build()
        return retrofit.create<MeetApi>(
            MeetApi::class.java)
    }
}